lib/src/auth/firebase_auth.dart

```diff
@@ -34,15 +44,27 @@ class FirebaseAuth {
   AuthGateway _authGateway;
   UserGateway _userGateway;
 
-  FirebaseAuth(this.apiKey, TokenStore tokenStore, {this.httpClient})
-      : assert(apiKey.isNotEmpty),
+  FirebaseAuth(
+    this.apiKey,
+    TokenStore tokenStore, {
+    String authGatewayUrl,
```

Comment:
```suggestion
    String authGatewayUrl = AuthConstants.authGatewayUrl,
```
Should we add default value here as well?